src/widgets/accountRecovery/accountRecoveryWidget.tsx

```diff
@@ -0,0 +1,217 @@
+            <PasskeysExplanation/>
+            <DeviceInputForm
+                handler={handleSubmit}
+                onSuccess={handleSuccess}
+                onError={onError}
+            />
+            {allowCreatePassword &&
+                <Alternative>
+                    <Separator text={i18n('or')} />
+                    <Intro><Link target="new-password">Create a new password</Link></Intro>
+                </Alternative>
+            }
+        </div>
+    )
+}
+
+interface SuccessViewProps {
+    loginLink?: string
+}
+
+const PasskeySuccessView = ({loginLink}: SuccessViewProps) => {
+    const i18n = useI18n()
+    return (
+        <div>
+            <Heading>{i18n('passkeyReset.successTitle')}</Heading>
+            <Info>{i18n('passkeyReset.successMessage')}</Info>
+            {loginLink && (
+                <Info>
+                    <Link href={loginLink}>{i18n('passkeyReset.loginLink')}</Link>
+                </Info>
+            )}
+        </div>
+    )
+}
+
+const PasswordSuccessView = ({loginLink}: SuccessViewProps) => {
+    const i18n = useI18n()
+    return (
+        <div>
+            <Heading>{i18n('passwordReset.successTitle')}</Heading>
+            <Info>{i18n('passwordReset.successMessage')}</Info>
+            {loginLink && (
+                <Info>
+                    <Link href={loginLink}>{i18n('passwordReset.loginLink')}</Link>
+                </Info>
+            )}
+        </div>
+    )
+}
+
+export const NewPasswordView = ({
+                             authentication,
+                             onSuccess = () => {},
+                             onError = () => {},
+                             showLabels = false,
+                         }: PropsWithAuthentication<MainViewProps>) => {
+    const coreClient = useReachfive()
+    const i18n = useI18n()
+    const { goTo } = useRouting()
+
+    const handleSubmit = ({ password }: PasswordEditorFormData) => {
+        return coreClient.updatePassword({
+            password,
+            ...authentication
+        });
+    };
+
+    const handleSuccess = () => {
+        onSuccess();
+        goTo('password-success');
+    };
+
+    return (
+        <div>
+            <Heading>{i18n('accountRecovery.password.title')}</Heading>
+            <Info>{i18n('passwordReset.intro')}</Info>
+            <PasswordEditorForm
+                handler={handleSubmit}
+                showLabels={showLabels}
+                onSuccess={handleSuccess}
+                onError={onError} />
+            <Alternative>
+                <Link target="new-passkey">Back</Link>
+            </Alternative>
+        </div>
+    )
+}
+
+const resolveCode = () => {
+    const qs = (window.location.search && window.location.search.length)
+        ? window.location.search.substr(1)
+        : '';
+    const {verificationCode, email, clientId} = parseQueryString(qs)
+    return {authentication: { verificationCode, email, clientId } as Authentication};
+};
```

Comment:
```suggestion
const resolveCode = (): Authentication => {
    const qs = (window.location.search && window.location.search.length)
        ? window.location.search.substr(1)
        : '';
    const { verificationCode, email, clientId } = parseQueryString(qs)
    return { authentication: { verificationCode, email, clientId }};
};
```
Mais problème, `Authentication` ne défini pas de propriété `authentication`. 
Donc soit tu change le type de retour par quelque chose comme `{ authentication: Authentication }`, mais c'est pas terrible.
Soit tu change l'utilisation de `resolveCode` par quelque chose du genre : 
```typescript
{...{ authentication: resolveCode() }}
```